core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -186,20 +187,23 @@ static paddr_t core_mmu_get_ul1_ttb_pa(void)
 	return pa;
 }
 
-static void *core_mmu_alloc_l2(struct tee_mmap_region *mm)
+static void *core_mmu_alloc_l2(size_t size)
```

Comment:
I see this as a nice small increment that solves the problem of supplying more than one page table.
Dynamic allocation of these tables is a different problem and I actually expect that without much effort we can design the system in a way that we'll never need that. Kernel mode currently needs kernel mapped shared memory in two cases, when loading a TA and when working with secure storage. Secure storage only needs 4-8 KiB of memory so that's nothing and loading TAs can used quite a lot as the entire TA is loaded at once. If we change the way we load TAs to only load parts at a time we can limit the amount shared memory that need to be mapped at the same time.